[Bug]: Discord Dynamica

### Panel Version

1.11.3

### Wings Version

1.11.7

### Service

bots/discord/dynamica

### Modified

Yes, I modified the egg (will provide details below)

### Expected Behavior

Removed the limit of 18 characters for the bot client ID, as discord now has above 18 characters for client IDs.

`|size:18` is what was removed

### Actual Behavior

Server doesn't start up. Heres the error

```
container@pterodactyl~ Server marked as starting...
[Pterodactyl Daemon]: Pulling Docker container image, this could take a few minutes to complete...
Pulling from parkervcp/yolks 
Digest: sha256:42339ed43474d121a107bfb856db9fb466dfffff681aca4b53e02fe2f12b6700 
Status: Image is up to date for ghcr.io/parkervcp/yolks:nodejs_16 
[Pterodactyl Daemon]: Finished pulling Docker container image
v16.20.2
:/home/container$ yarn deploy && yarn prisma migrate deploy && yarn start
yarn run v1.22.19
$ echo "This script is no longer used. All you need to do is run yarn start and deployment happens automatically."
This script is no longer used. All you need to do is run yarn start and deployment happens automatically.
Done in 0.13s.
yarn run v1.22.19
$ /home/container/node_modules/.bin/prisma migrate deploy
Prisma schema loaded from prisma/schema.prisma
Datasource "db": SQLite database "db.sqlite" at "file:/home/container/dynamica/db.sqlite"
14 migrations found in prisma/migrations
No pending migrations to apply.
Done in 5.34s.
yarn run v1.22.19
$ node --enable-source-maps .
node:internal/modules/cjs/loader:435
      throw err;
      ^
Error: Cannot find module '/home/container/dist/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:427:19)
    at Function.Module._findPath (node:internal/modules/cjs/loader:640:18)
    at resolveMainPath (node:internal/modules/run_main:19:25)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:75:24)
    at node:internal/main/run_main_module:22:47 {
  code: 'MODULE_NOT_FOUND',
  path: '/home/container/package.json',
  requestPath: '/home/container'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
container@pterodactyl~ Server marked as offline...
[Pterodactyl Daemon]: ---------- Detected server process in a crashed state! ----------
[Pterodactyl Daemon]: Exit code: 1
[Pterodactyl Daemon]: Out of memory: false
[Pterodactyl Daemon]: Aborting automatic restart, last crash occurred less than 60 seconds ago.
```

### Steps To Reproduce

Change the variable for CLIENT_ID under rules to only have the following:

Old: `"rules": "required|string|size:18"`
New: `"rules": "required|string"`

Create a new server
Enter a bot token and client id into the variables
Leave everything else default
Start server installation process
Boot server up
Error

### Install logs

https://pteropaste.com/eqvwzswsjdjc

--- a/bots/discord/dynamica/README.md
+++ b/bots/discord/dynamica/README.md
@@ -1,5 +1,8 @@
 # Dynamica
 
-### Their [Github](https://github.com/dynamicabot/dynamica)
+## Their [Github](https://github.com/dynamicabot/dynamica-v2)
 
-This is a discord bot for dynamically managing voice channels.
+An easy-to-use dynamic voice channel bot.
+
+## Port
+You will manual have to change the web port in the src/main.ts on line 24.

--- a/bots/discord/dynamica/egg-dynamica.json
+++ b/bots/discord/dynamica/egg-dynamica.json
@@ -1,28 +1,28 @@
 {
     "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
     "meta": {
-        "version": "PTDL_v1",
+        "version": "PTDL_v2",
         "update_url": null
     },
-    "exported_at": "2021-12-13T10:34:52+11:00",
+    "exported_at": "2023-08-18T10:12:08+00:00",
     "name": "Dynamica",
     "author": "sebastian.pietschner@gmail.com",
     "description": "The dynamica discord bot",
     "features": null,
-    "images": [
-        "ghcr.io\/parkervcp\/yolks:nodejs_16"
-    ],
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:nodejs_16": "ghcr.io\/parkervcp\/yolks:nodejs_16"
+    },
     "file_denylist": [],
-    "startup": "yarn deploy && yarn prisma migrate deploy && yarn start",
+    "startup": "yarn prisma migrate deploy && yarn start",
     "config": {
         "files": "{}",
-        "startup": "{\r\n    \"done\": \"Ready! Logged in as \"\r\n}",
+        "startup": "{\r\n    \"done\": \"Bot logged in\"\r\n}",
         "logs": "{}",
         "stop": "^C"
     },
     "scripts": {
         "installation": {
-            "script": "apt-get update\r\napt-get install openssl git -y -qq\r\ncd \/mnt\/server\r\nif [[ -d .git\/ ]]; then\r\n    git pull\r\nelse\r\n    git clone https:\/\/github.com\/dynamicabot\/dynamica.git .;\r\nfi;\r\nyarn install --frozen-lockfile",
+            "script": "apt-get update\r\napt-get install openssl git -y -qq\r\ncd \/mnt\/server\r\nif [[ -d .git\/ ]]; then\r\n    git pull\r\nelse\r\n    git clone https:\/\/github.com\/dynamicabot\/dynamica-v2.git .;\r\nfi;\r\nyarn install --frozen-lockfile",
             "container": "node:lts-buster-slim",
             "entrypoint": "bash"
         }
@@ -35,7 +35,8 @@
             "default_value": "file:\/home\/container\/dynamica\/db.sqlite",
             "user_viewable": true,
             "user_editable": false,
-            "rules": "required|string"
+            "rules": "required|string",
+            "field_type": "text"
         },
         {
             "name": "Token",
@@ -44,7 +45,8 @@
             "default_value": "",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "required|string"
+            "rules": "required|string",
+            "field_type": "text"
         },
         {
             "name": "Client ID",
@@ -53,7 +55,8 @@
             "default_value": "",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "required|string|size:18"
+            "rules": "required|string",
+            "field_type": "text"
         }
     ]
 }

--- a/bots/discord/dynamica/egg-dynamica.json
+++ b/bots/discord/dynamica/egg-dynamica.json
@@ -4,26 +4,26 @@
         "version": "PTDL_v2",
         "update_url": null
     },
-    "exported_at": "2023-08-18T10:12:08+00:00",
+    "exported_at": "2023-08-18T17:01:43+02:00",
     "name": "Dynamica",
     "author": "sebastian.pietschner@gmail.com",
-    "description": "The dynamica discord bot",
+    "description": "An easy-to-use dynamic voice channel bot.",
     "features": null,
     "docker_images": {
-        "ghcr.io\/parkervcp\/yolks:nodejs_16": "ghcr.io\/parkervcp\/yolks:nodejs_16"
+        "Nodejs 18": "ghcr.io\/parkervcp\/yolks:nodejs_18"
     },
     "file_denylist": [],
-    "startup": "yarn prisma migrate deploy && yarn start",
+    "startup": "npx prisma migrate deploy && yarn start",
     "config": {
         "files": "{}",
-        "startup": "{\r\n    \"done\": \"Bot logged in\"\r\n}",
+        "startup": "{\r\n    \"done\": \"Bot logged in as \"\r\n}",
         "logs": "{}",
-        "stop": "^C"
+        "stop": "^^C"
     },
     "scripts": {
         "installation": {
-            "script": "apt-get update\r\napt-get install openssl git -y -qq\r\ncd \/mnt\/server\r\nif [[ -d .git\/ ]]; then\r\n    git pull\r\nelse\r\n    git clone https:\/\/github.com\/dynamicabot\/dynamica-v2.git .;\r\nfi;\r\nyarn install --frozen-lockfile",
-            "container": "node:lts-buster-slim",
+            "script": "#!\/bin\/bash\r\n\r\napt update\r\napt -y install tar curl jq git\r\n\r\nmkdir -p \/mnt\/server\r\ncd \/mnt\/server\r\n\r\nGITHUB_PACKAGE=\"DynamicaBot\/Dynamica-V2\"\r\n\r\n# Find GitHub download URL for Muse for specified version\r\nLATEST_JSON=$(curl --silent \"https:\/\/api.github.com\/repos\/${GITHUB_PACKAGE}\/releases\/latest\")\r\nRELEASES=$(curl --silent \"https:\/\/api.github.com\/repos\/${GITHUB_PACKAGE}\/releases\")\r\n\r\nif [ -z \"${VERSION}\" ] || [ \"${VERSION}\" == \"latest\" ]; then\r\n    DOWNLOAD_URL=$(echo ${LATEST_JSON} | jq -r '.tarball_url')\r\nelse\r\n    VERSION_CHECK=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .tag_name')\r\n    if [ \"${VERSION}\" == \"${VERSION_CHECK}\" ]; then\r\n        DOWNLOAD_URL=$(echo ${RELEASES} | jq -r --arg VERSION \"${VERSION}\" '.[] | select(.tag_name==$VERSION) | .tarball_url')\r\n    else\r\n        echo -e \"Could not find version \\\"${VERSION}\\\" of Muse. Defaulting to latest release...\"\r\n        DOWNLOAD_URL=$(echo ${LATEST_JSON} | jq -r '.tarball_url')\r\n    fi\r\nfi\r\n\r\ncurl -sSL -o dynamica.tar.gz ${DOWNLOAD_URL}\r\ntar xvf dynamica.tar.gz --strip-components=1\r\nrm dynamica.tar.gz\r\n\r\n#curl -sSL -o tsconfig.json https:\/\/pteropaste.com\/xqhjja443pbm\r\n\r\nyarn add tsup --dev\r\nyarn install  --frozen-lockfile\r\n\r\nyarn generate\r\nyarn build\r\n\r\nrm -rf .github\/ .vscode\/ Dockerfile partial.Dockerfile entrypoint.sh dynamica-egg.json\r\n\r\n\r\n## install end\r\necho \"-----------------------------------------\"\r\necho \"Installation completed...\"\r\necho \"-----------------------------------------\"",
+            "container": "node:18-bullseye-slim",
             "entrypoint": "bash"
         }
     },
@@ -49,13 +49,53 @@
             "field_type": "text"
         },
         {
-            "name": "Client ID",
-            "description": "The client ID of your discord bot.",
-            "env_variable": "CLIENT_ID",
+            "name": "Guild ID",
+            "description": "Add only if using the bot in a single guild.",
+            "env_variable": "GUILD_ID",
             "default_value": "",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "required|string",
+            "rules": "string|nullable",
+            "field_type": "text"
+        },
+        {
+            "name": "Sentry DSN",
+            "description": "The DSN for Sentry.",
+            "env_variable": "SENTRY_DSN",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "string|nullable",
+            "field_type": "text"
+        },
+        {
+            "name": "MQTT URL",
+            "description": "The URL for the MQTT broker.",
+            "env_variable": "MQTT_URL",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "string|nullable",
+            "field_type": "text"
+        },
+        {
+            "name": "MQTT Username",
+            "description": "The username for the MQTT broker.",
+            "env_variable": "MQTT_USER",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "string|nullable",
+            "field_type": "text"
+        },
+        {
+            "name": "MQTT Password",
+            "description": "The password for the MQTT broker.",
+            "env_variable": "MQTT_PASS",
+            "default_value": "",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "string|nullable",
             "field_type": "text"
         }
     ]

